opennms-config/src/test/java/org/opennms/netmgt/config/WmiPeerFactoryTest.java

```diff
@@ -28,43 +28,61 @@
 
 package org.opennms.netmgt.config;
 
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
+import org.opennms.features.config.dao.impl.util.JaxbXmlConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
 import java.io.IOException;
 import java.net.InetAddress;
 
-import junit.framework.TestCase;
-
-import org.opennms.core.test.ConfigurationTestUtils;
+import static org.junit.Assert.assertEquals;
 
 /**
  * JUnit tests for the configureSNMP event handling and optimization of
  * the SNMP configuration XML.
- * 
- * @author <a href="mailto:david@opennms.org>David Hustace</a>
  *
+ * @author <a href="mailto:david@opennms.org>David Hustace</a>
  */
-public class WmiPeerFactoryTest extends TestCase {
-    private WmiPeerFactory getFactory(String amiConfigXml) throws IOException {
-        WmiPeerFactory factory = new WmiPeerFactory(ConfigurationTestUtils.getResourceForConfigWithReplacements(amiConfigXml));
-        factory.afterPropertiesSet();
-        return factory;
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+        "classpath:applicationContext-mockWmiPeerFactory.xml"
+
+})
+public class WmiPeerFactoryTest {
+
+    @Autowired
+    private WmiPeerFactory factory;
+
+    private void updateConfig(String amiConfigXml) {
+        try {
+            JaxbXmlConverter converter = new JaxbXmlConverter("wmi-config.xsd", "wmi-config", null);
+            String json = converter.xmlToJson(amiConfigXml);
+            factory.updateConfig(json);
+            factory.reload();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
rather do throw the exception since we are in a test?

Reply:
removed